spring-integration-ip/src/test/java/org/springframework/integration/ip/udp/UdpChannelAdapterTests.java

```diff
@@ -275,40 +273,21 @@ public void testMulticastReceiver() throws Exception {
 		adapter.stop();
 	}
 
-	private String checkMulticast() throws Exception {
-		String nic = SocketTestUtils.chooseANic(true);
-		if (nic == null) {	// no multicast support
-			LogFactory.getLog(this.getClass()).info("No Multicast support; test skipped");
-			return null;
-		}
-		try {
-			MulticastSocket socket = new MulticastSocket();
-			socket.joinGroup(InetAddress.getByName("225.6.7.9"));
-			socket.close();
-		}
-		catch (Exception e) {
-			LogFactory.getLog(this.getClass()).info("No Multicast support; test skipped");
-		}
-		return nic;
-	}
-
 	@SuppressWarnings("unchecked")
 	@Test
 	public void testMulticastSender() throws Exception {
-		System.setProperty("java.net.preferIPv4Stack", "true");
 		QueueChannel channel = new QueueChannel(2);
 		int port = SocketUtils.findAvailableUdpSocket();
-		UnicastReceivingChannelAdapter adapter = new MulticastReceivingChannelAdapter("225.6.7.9", port);
+		UnicastReceivingChannelAdapter adapter =
+				new MulticastReceivingChannelAdapter(this.multicastRule.getGroup(), port);
```

Comment:
Yes, I know. Does it hurt, having the fact that we are in a closed contex within each test in this class?